src/libraries/System.Private.CoreLib/src/System/UInt128.cs

```diff
@@ -1103,6 +1104,20 @@ public static UInt128 Log2(UInt128 value)
                     // left and right are both uint64
                     return left._lower / right._lower;
                 }
+                else if (X86Base.X64.IsSupported)
+                {
+                    ulong highRes = 0ul;
+                    ulong remainder = left._upper;
+
+#pragma warning disable CA2252 // This API requires opting into preview features
+                    if (remainder >= right._lower)
+                    {
+                        (highRes, remainder) = X86Base.X64.DivRem(left._upper, 0, right._lower);
+                    }
+
+                    return new UInt128(highRes, X86Base.X64.DivRem(left._lower, remainder, right._lower).Quotient);
+#pragma warning restore CA2252 // This API requires opting into preview features
+                }
```

Comment:
If you find it more readable then the code can be changed to the below.

### if/else

```suggestion
                else if (X86Base.X64.IsSupported)
                {
#pragma warning disable CA2252 // This API requires opting into preview features
                    if (left._upper >= right._lower)
                    {
                        (ulong highRes, ulong remainder) =  X86Base.X64.DivRem(left._upper, 0, right._lower);
                        return new UInt128(highRes, X86Base.X64.DivRem(left._lower, remainder, right._lower).Quotient);
                    }
                    else
                    {
                        return X86Base.X64.DivRem(left._lower, left._upper, right._lower).Quotient;
                    }
#pragma warning restore CA2252 // This API requires opting into preview features
                }
```

### if else with Math.DivRem

Math.DivRem could also be used for the upper bit, it is slightly slower for the general case, but should be a slight improvement for constant divisor.  (it might have been maybe ~one percent for modified example which is found below: but the values changed to much to be certain) 

```suggestion
                else if (X86Base.X64.IsSupported)
                {
#pragma warning disable CA2252 // This API requires opting into preview features
                    if (left._upper >= right._lower)
                    {
                        (ulong highRes, ulong remainder) = Math.DivRem(left._upper, right._lower);
                        return new UInt128(highRes, X86Base.X64.DivRem(left._lower, remainder, right._lower).Quotient);
                    }
                    else
                    {
                        return X86Base.X64.DivRem(left._lower, left._upper, right._lower).Quotient;
                    }
#pragma warning restore CA2252 // This API requires opting into preview features
                }
```

<details>
<summary>modified example</summary>

``` 

for (; ; )
{
    sw.Restart();
    UInt128 sum = 0;
    UInt128 min = new UInt128(1ul, 0);
    UInt128 max = new UInt128(1ul, 100_000_000);

    for (UInt128 i = min; i < max; i++) 
        sum += (UInt128)i / (UInt128)3;
    Console.WriteLine(sw.ElapsedMilliseconds);
}
```
</details>